tests/vmi_multus_test.go

```diff
@@ -1084,6 +1110,16 @@ func runSafeCommand(vmi *v1.VirtualMachineInstance, command string) error {
 	}, 15)
 }
 
+func getSRIOVInterfaceNameByMAC(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachineInstance, mac string) string {
```

Comment:
You could return an error from this function with the message "Could not find and interface with MAC address %s", 
then in the caller function return an error that warping this function error.
Its easier to give constructive error message this way IMO.
for example: 
 ```
configSRIOVInterface(...) {
...
    ifaceName, err := getSRIOVInterfaceNameByMAC(...)
    if err != nil {
       return fmt.Errorf("Could not configure SRIOV interface with MAC address '%s', %v", macAddress, err)
   }
....
}
```

Reply:
This is what I had originally [here](https://github.com/kubevirt/kubevirt/compare/528149a9816cd07c2ace101ab4d4e3871ad6063c..dd08c2269797b2c4f676ac18afa6f916c13750cd) , but changed it as @oshoval  asked [here](https://github.com/kubevirt/kubevirt/pull/4001#discussion_r512801229)

I think returning empty string make it a little more simpler, but don't mind that much.
what do you think @oshoval  @ormergi ?

Reply:
a function that fails on one thing only, and the name of it imply what it should do so good, i dont think we need another string of error here, as i thought before

Reply:
No matter how good the function name document the code, you wont see it on the logs.
Eventually you will start to debug it and it will take you more time to find where it failed no mater how simple the function is.
In this case `configSRIOVInterface` could fail due to other reasons, and if we use helper function let them propagate the error.

Reply:
we talk on getSRIOVInterfaceNameByMAC
and every caller already check the return value and have a print
so even less we need it imo

Reply:
But it will be logged from the assertion: 
like : 
```
interfaceName := getSRIOVInterfaceNameByMAC(virtClient, vmi, mac)
Expect(interfaceName).NotTo(BeEmpty(), "no SRIOV interface on vmi %s with mac: %s", vmi.Name, mac)
```

Reply:
I prefer to return an error with informative message instead of an empty string.
Eventually we will need to explain why it is empty in the caller function, which is out of the scope of the caller.

About the assertion, I am not quite sure what you mean, but it is better to keep the assertion on the test code (I mean not in helpers)

Reply:
eventually I am back to the original decision , by making the helper to return an error and a string. I agree with @ormergi  that it is not the caller job to log what went wrong with the helper. Whoever uses that helper should just assert on the error.